bench-tps/src/bench.rs

```diff
@@ -395,10 +440,69 @@ fn generate_system_txs(
         .collect()
 }
 
-fn generate_txs(
+fn get_nonce_blockhash<T: 'static + BenchTpsClient + Send + Sync>(
+    client: Arc<T>,
+    nonce_account_pubkey: Pubkey,
+) -> Hash {
+    let nonce_account = client
+        .get_account(&nonce_account_pubkey)
+        .unwrap_or_else(|error| panic!("{:?}", error));
+    let nonce_data = nonce_utils::data_from_account(&nonce_account)
+        .unwrap_or_else(|error| panic!("{:?}", error));
+    nonce_data.blockhash()
+}
+
+fn generate_nonced_system_txs<T: 'static + BenchTpsClient + Send + Sync>(
+    client: Arc<T>,
+    source: &[&Keypair],
+    dest: &VecDeque<&Keypair>,
+    source_nonce: &[&Keypair],
+    dest_nonce: &VecDeque<&Keypair>,
+    reclaim: bool,
+) -> Vec<(Transaction, u64)> {
+    let length = source.len();
+    let mut transactions: Vec<(Transaction, u64)> = Vec::with_capacity(length);
+    for i in 0..length {
+        let (from, to, nonce, nonce_blockhash) = if !reclaim {
+            (
+                source[i],
+                dest[i],
+                source_nonce[i],
+                get_nonce_blockhash(client.clone(), source_nonce[i].pubkey()),
+            )
+        } else {
+            (
+                dest[i],
+                source[i],
+                dest_nonce[i],
+                get_nonce_blockhash(client.clone(), dest_nonce[i].pubkey()),
+            )
+        };
+
+        transactions.push((
+            system_transaction::nonced_transfer(
+                from,
+                &to.pubkey(),
+                1,
+                &nonce.pubkey(),
+                from,
+                nonce_blockhash,
+            ),
+            timestamp(),
+        ));
+    }
+    // current timestamp to avoid filtering out some transactions if they are too old
+    let t = timestamp();
+    for mut tx in &mut transactions {
+        tx.1 = t;
+    }
```

Comment:
small point: It's too bad to have to do this, since the whole point of the nonce transactions is that they can't be too old!  Up to you, but this could be made more generally useful for nonces if `SharedTransactions` were instead `Arc<RwLock<VecDeque<Vec<(Transaction, Option<u64>)>>>>`, and the old check is only done if a timestamp is `Some`. What do you think?

Reply:
Generally, make sense. Kind of long preexisting type definition (`Arc<....>`). I now think how to simplify by introducing type alias for `Vec<(Transaction, Option<u64>)>`. Like `TransactionsWithTimestamp` (sound cumbersome but didn't came up with better name)